salesforce/client.bal

```diff
@@ -69,6 +65,250 @@ public isolated client class Client {
+        string path = utils:prepareUrl([API_BASE_PATH, SOBJECTS, sObjectName, id]);
+        return check self.salesforceClient->delete(path);
+    }
+
+    # Executes the specified SOQL query.
+    #
+    # + soql - SOQL query
+    # + returnType - The payload, which is expected to be returned after data binding.
+    # + return - `stream<{returnType}, error?>` if successful. Else, the occurred `error`.
+    @display {label: "Get Query Result"}
+    isolated remote function query(@display {label: "SOQL Query"} string soql, typedesc<record {}> returnType = <>)
+                                    returns @display {label: "SOQL Result"} stream<returnType, error?>|error = @java:Method {
+        'class: "io.ballerinax.salesforce.ReadOperationExecutor",
+        name: "getQueryResult"
+    } external;
+
+    private isolated function processGetQueryResult(typedesc<record {}> returnType, string receivedQuery)
+                                                    returns stream<record {}, error?>|error {
+        string path = utils:prepareQueryUrl([API_BASE_PATH, QUERY], [Q], [receivedQuery]);
+        SOQLQueryResultStream objectInstance = check new (self.salesforceClient, path);
+        stream<record {}, error?> finalStream = new (objectInstance);
+        return self.streamConverter(finalStream, returnType);
+    }
+
+    # Executes the specified SOSL search.
+    #
+    # + sosl - SOSL search query
+    # + returnType - The payload, which is expected to be returned after data binding.
+    # + return - `stream<{returnType}, error?>` record if successful. Else, the occurred `error`.
+    @display {label: "SOSL Search"}
+    isolated remote function search(@display {label: "SOSL Search Query"} string sosl, typedesc<record {}> returnType = <>)
+                                    returns @display {label: "SOSL Result"} stream<returnType, error?>|error = @java:Method {
+        'class: "io.ballerinax.salesforce.ReadOperationExecutor",
+        name: "searchSOSLString"
+    } external;
+
+    private isolated function processSearchSOSLString(typedesc<record {}> returnType, string searchString)
+                                                    returns stream<record {}, error?>|error {
+        string path = utils:prepareQueryUrl([API_BASE_PATH, SEARCH], [Q], [searchString]);
+
+        SOSLSearchResult objectInstance = check new (self.salesforceClient, path);
+        stream<record {}, error?> finalStream = new (objectInstance);
+        return self.streamConverter(finalStream, returnType);
+    }
+
+    // External function for the conversion of stream
+    isolated function streamConverter(stream<record {}, error?> data, typedesc<record {}> returnType) returns
+    stream<record {}, error?>|error = @java:Method {
+        'class: "io.ballerinax.salesforce.ReadOperationExecutor"
+    } external;
+
     ///////////////////////////////////////////// DEPRECATED ///////////////////////////////////////////////////////////
 
     //Describe SObjects
```

Comment:
Check and update the deprecated comment in all deprecated operations.

Reply:
Updated in https://github.com/ballerina-platform/module-ballerinax-sfdc/commit/0997adbe6d633d3c7bc158f492d44430435bd7fb